comb_spec_searcher/strategies/constructor/cartesian.py

```diff
@@ -334,12 +334,12 @@ def random_sample_sub_objects(
             total += tmp
             if random_choice <= total:
                 extra_parameters = self.get_extra_parameters(child_parameters)
-                break
-        assert extra_parameters is not None
-        return tuple(
-            subsampler(n=extra_params.pop("n"), **extra_params)
-            for subsampler, extra_params in zip(subsamplers, extra_parameters)
-        )
+                assert extra_parameters is not None
+                return tuple(
+                    subsampler(n=extra_params.pop("n"), **extra_params)
+                    for subsampler, extra_params in zip(subsamplers, extra_parameters)
+                )
+        raise Exception("Function did not return")
```

Comment:
It is bad practice to raise a plain exception. Maybe RuntimeError would be more appropriate here

Reply:
Fixed